src/components/backend-ai-login.ts

```diff
@@ -1979,6 +1979,25 @@ export default class BackendAILogin extends BackendAIPage {
       : password.setAttribute('type', 'password');
   }
 
+  private _validateInput() {
+    // Email validation pattern
+    const emailPattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+    // Username validation pattern
+    const usernamePattern = /^[a-zA-Z0-9._-]{3,20}$/;
```

Comment:
Can you include the reference to this rule in the description?

Reply:
Agree. I found that username is allowed to 64 characters, but is it really restricted to 20 characters?

Reply:
I couldn't find rules for the input email and username, so I set them arbitrarily: for email, I searched a popular expression and applied it, and for username, I removed the max_length rule and changed it to apply only min_length (max_length is defined in mwc-textfield). 

In the description, I've written down what type of input each regular expression accepts.